### PR TITLE
goforit: Use a context instead of globals

### PR DESCRIPTION
**Pull request by vasi-stripe**
This should allow use of goforit even in code that will be under
test.

### CODE REVIEW

File: examples_test.go [hunk 1]
```diff
@@ -9,13 +10,13 @@ func Example() {
 	// flags.csv contains comma-separated flag names and sample rates.
 	// See: fixtures/flags_example.csv
 	backend := BackendFromFile("flags.csv")
-	Init(30*time.Second, backend)
+	_, ctx := Init(context.Background(), 30*time.Second, backend)
```
Comment (ChimeraCoder) [inline examples_test.go]: Context is generally intended to be request-level, which means that each request would require initializing separately (and therefore a separate backend and ticker).

I don't think we actually need to pass a fresh context into the initialization, though - we could use the context to override and/or cache the values, but still otherwise creating the backend outside of the request lifecycle.

File: flags_test.go [hunk 2]
```diff
@@ -73,12 +74,12 @@ func TestEnabled(t *testing.T) {
 	ticker := Init(DefaultInterval, backend)
 	defer ticker.Stop()
 
-	assert.False(t, Enabled("go.sun.money"))
-	assert.True(t, Enabled("go.moon.mercury"))
+	assert.False(t, Enabled(context.Background(), "go.sun.money"))
```
Comment (ChimeraCoder) [inline flags_test.go]: Can we also add a test for a nil context?

File: examples_whitebox_test.go [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 	backend := goforit.BackendFromFile("flags.csv")
 	goforit.Init(30*time.Second, backend)
 
-	if goforit.Enabled("go.sun.mercury") {
+	if goforit.Enabled(ctx, "go.sun.mercury") {
 		fmt.Println("The go.sun.mercury feature is enabled for 100% of requests")
 	}
 
-	if goforit.Enabled("go.stars.money") {
+	if goforit.Enabled(ctx, "go.stars.money") {
```
Comment (ChimeraCoder) [inline examples_whitebox_test.go]: Let's also add an example of using this without context (since passing in a nil context should be a safe operation).

**Comment by ChimeraCoder** [comment]
left two comments - looks good to me after that!

**Comment by vasi-stripe** [comment]
I think the comments are addressed, we have tests/examples using nil.

**Comment by ChimeraCoder** [comment]
lgtm
